### PR TITLE
Some PRSTUX changes

### PR DESCRIPTION
**Pull request by v01d**
I had this locally and missed sending a PR. There are some small wifi/suspend related fixes and this also completes support for detecting chargin mode to exit and inhibit auto-suspend.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/4567)

### CODE REVIEW

**Comment by v01d** [comment]
For the first change, should I just call "reconnectOrShowNetworkMenu"? If so, I can test that and update the PR with that. For the second one, I haven't yet managed to write a script to safely restore wifi state after susped (that call is commented for prstux), so I would skip that for now.

**Comment by pazos** [comment]
@v01d: both changes should work as is as long as your device is using wpasupplicant (directly or via some weird wrapper).

The first change is, like you said, replace showNetworkMenu with reconnectOrShowNetworkMenu. 
The second change is to copy everything in https://github.com/koreader/koreader/pull/4463/files.

Anyways, you can do it later if you wish.

File: frontend/device/sony-prstux/device.lua [hunk 1]
```diff
@@ -1,6 +1,7 @@
 local Generic = require("device/generic/device") -- <= look at this file!
 local logger = require("logger")
 local TimeVal = require("ui/timeval")
+local PluginShare = require("pluginshare")
```
Comment (Frenzie) [inline frontend/device/sony-prstux/device.lua]: Oh, could you please also sort these as per the coding style?

https://github.com/koreader/koreader-base/wiki/Coding-style#order-of-requires

**Comment by pazos** [comment]
Apparently I commited once to the branch (before cherry-picking) and now I'm not able to push the changes. Sorry. I will open a new PR

**Pull request closed without merging**